samples/dnn/text_detection.cpp

```diff
@@ -72,23 +87,69 @@ int main(int argc, char** argv)
         }
 
         blobFromImage(frame, blob, 1.0, Size(inpWidth, inpHeight), Scalar(123.68, 116.78, 103.94), true, false);
-        net.setInput(blob);
-        net.forward(outs, outNames);
+        net_detector.setInput(blob);
+        net_detector.forward(outs, outNames);
 
         Mat scores = outs[0];
         Mat geometry = outs[1];
 
         // Decode predicted bounding boxes.
         std::vector<RotatedRect> boxes;
         std::vector<float> confidences;
-        decode(scores, geometry, confThreshold, boxes, confidences);
+        decode_bounding_boxes(scores, geometry, confThreshold, boxes, confidences);
 
         // Apply non-maximum suppression procedure.
         std::vector<int> indices;
         NMSBoxes(boxes, confidences, confThreshold, nmsThreshold, indices);
 
-        // Render detections.
         Point2f ratio((float)frame.cols / inpWidth, (float)frame.rows / inpHeight);
+
+        // Render text.
+        for (size_t i = 0; i < indices.size() && !model_recognition.empty(); ++i)
+        {
+            RotatedRect& box = boxes[indices[i]];
+
+            float angle = box.angle;
+
+            Size box_size = box.size;
+            Point2f box_center = box.center;
+
+            box_size.width *= ratio.x;
+            box_size.height *= ratio.y;
+
+            box_center.x *= ratio.x;
+            box_center.y *= ratio.y;
+
+
+            // Rotate and crop rectangle with text
+            Mat rotation_matrix = getRotationMatrix2D(box_center, angle, 1.0);
+            Mat rotated;
+            warpAffine(frame, rotated, rotation_matrix, frame.size(), INTER_CUBIC);
+            Mat cropped;
+            getRectSubPix(rotated, box_size, box_center, cropped);
+
+
+            cropped.convertTo(cropped, CV_32F, 1.0/255);
+            cropped -= 0.5;
+            cropped /= 0.5;
```

Comment:
Please use blobFeomImage arguments for preprocessing

Reply:
`blobFromImage(cropped, 1.0/127.5, Size(100,32), Scalar(127), false, false);`

Reply:
Even without `false, false` due they are by default.